### PR TITLE
fix: use correct external IP on EC2 instances

### PR DESCRIPTION
**Pull request by vmunich**
AWS doesn't expose the instance's external IP on `ifconfig`, instead it is exposed through [Instance Metadata](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html).This fix will use the correct external IP address when using Deployer on EC2 instances.

## What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Does this PR release a new version?

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: lib/utils.sh [hunk 1]
```diff
@@ -2,12 +2,16 @@
 
 get_ip()
 {
-    IP=$(sudo ifconfig | fgrep "inet " | egrep -v "inet (addr:)?(127|192)\." | egrep -o "inet (addr:)?([0-9]+\.?){4}" | egrep -o "([0-9]+\.?){4}" | head -n 1)
-
-    if [ ! -z "$IP" ]; then
-        echo "$IP"
+    if AWS_IP=$(curl -f -s --connect-timeout 5 http://169.254.169.254/latest/meta-data/public-ipv4); then
```
Comment (alexbarnsley) [inline lib/utils.sh]: This URL doesn't seem to be loading for me, is it an Amazon-specific URL?

Comment (vmunich) [reply lib/utils.sh]: @alexbarnsley yes this is specific to AWS.

Comment (vmunich) [reply lib/utils.sh]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html

Comment (alexbarnsley) [reply lib/utils.sh]: Thanks for clarifying. It may be worth a comment, in case someone else tries the same thing

**Comment by alexbarnsley** [comment]
Not sure if you want to add that comment first though. I can re-approve if you do

**Comment by vmunich** [comment]
Added a comment 😄 @alexbarnsley

**Pull request merged**